packages/build/src/plugins/child/logic.js

```diff
@@ -1,22 +1,43 @@
 'use strict'
 
+const { pathToFileURL } = require('url')
+
 const { addTsErrorInfo } = require('./typescript')
 
 // Require the plugin file and fire its top-level function.
 // The returned object is the `logic` which includes all event handlers.
-const getLogic = function ({ pluginPath, inputs, tsNodeService }) {
-  const logic = requireLogic(pluginPath, tsNodeService)
+const getLogic = async function ({ pluginPath, inputs, tsNodeService }) {
+  const logic = await importLogic(pluginPath, tsNodeService)
   const logicA = loadLogic({ logic, inputs })
   return logicA
 }
 
-const requireLogic = function (pluginPath, tsNodeService) {
+const importLogic = async function (pluginPath, tsNodeService) {
   try {
-    // eslint-disable-next-line node/global-require, import/no-dynamic-require
-    return require(pluginPath)
+    // `ts-node` is not available programmatically for pure ES modules yet,
```

Comment:
have you tried running the binary with the `node --loader ts-node/esm`? Maybe this helps.

This should instruct node to use typescripts esm loader to run javascript/typescript code

Reply:
That's the plan! 
I have an issue at https://github.com/netlify/pod-workflow/issues/330 to implement `--loader ts-node/esm`.

Reply:
ahh cool awesome 😎